lib/acts-as-taggable-array-on/taggable.rb

```diff
@@ -35,6 +45,21 @@ def acts_as_taggable_array_on(tag_name, *)
         end
       end
       alias_method :taggable_array, :acts_as_taggable_array_on
+
+      def define_allowed_validations!(tag_name, allowed)
+        raise InvalidAllowListTypeError if !allowed.is_a?(Array)
+
+        remove_const "#{tag_name.upcase}_ALLOWED" if const_defined?("#{tag_name.upcase}_ALLOWED")
```

Comment:
I'm afraid, that my previous comment about constant was incorrectly phrased and understood. Sorry about that. 

Would be awesome not to make this impementation detail part of public api for this gem.  Can we hide this `allowed` variable inside a private method?

Reply:
I'm not most used to this meta-programming method definitions in ruby, but I tried to apply your suggestion here https://github.com/tmiyamon/acts-as-taggable-array-on/pull/40/commits/d5c44f160a9370319b76a1f6fcbf6694265040d3